zetaclient/chains/solana/observer/gas_price.go

```diff
@@ -0,0 +1,67 @@
+package observer
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/gagliardetto/solana-go/rpc"
+	"github.com/pkg/errors"
+
+	clienttypes "github.com/zeta-chain/zetacore/zetaclient/types"
+)
+
+// WatchGasPrice watches the gas price of the chain and posts it to the zetacore
+func (ob *Observer) WatchGasPrice(ctx context.Context) error {
+	// report gas price right away as the ticker takes time to kick in
+	err := ob.PostGasPrice(ctx)
+	if err != nil {
+		ob.Logger().GasPrice.Error().Err(err).Msgf("PostGasPrice error for chain %d", ob.Chain().ChainId)
+	}
+
+	// start gas price ticker
+	ticker, err := clienttypes.NewDynamicTicker(
+		fmt.Sprintf("Solana_WatchGasPrice_%d", ob.Chain().ChainId),
+		ob.GetChainParams().GasPriceTicker,
+	)
+	if err != nil {
+		return errors.Wrapf(err, "NewDynamicTicker error")
+	}
+	ob.Logger().GasPrice.Info().Msgf("WatchGasPrice started for chain %d with interval %d",
+		ob.Chain().ChainId, ob.GetChainParams().GasPriceTicker)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ticker.C():
+			if !ob.GetChainParams().IsSupported {
+				continue
+			}
+			err = ob.PostGasPrice(ctx)
+			if err != nil {
+				ob.Logger().GasPrice.Error().Err(err).Msgf("PostGasPrice error for chain %d", ob.Chain().ChainId)
+			}
+			ticker.UpdateInterval(ob.GetChainParams().GasPriceTicker, ob.Logger().GasPrice)
+		case <-ob.StopChannel():
+			ob.Logger().GasPrice.Info().Msgf("WatchGasPrice stopped for chain %d", ob.Chain().ChainId)
+			return nil
+		}
+	}
+}
+
+// PostGasPrice posts gas price to zetacore
+func (ob *Observer) PostGasPrice(ctx context.Context) error {
+	// get current slot
+	slot, err := ob.solClient.GetSlot(context.Background(), rpc.CommitmentConfirmed)
+	if err != nil {
+		return errors.Wrap(err, "GetSlot error")
+	}
+
+	// post gas price to zetacore
+	// FIXME: what's the fee rate of compute unit? How to query?
+	_, err = ob.ZetacoreClient().PostVoteGasPrice(ctx, ob.Chain(), 1, 0, slot)
```

Comment:
Keeping this comment as an anchor for visibility

Reply:
https://github.com/zeta-chain/node/pull/2560/commits/e88593e01650473035620204a07edaac44965157

Reply:
This is fixed during the implementation.